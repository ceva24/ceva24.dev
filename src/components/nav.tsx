import React from "react";
import { Link } from "gatsby";
import { useLocation } from "@gatsbyjs/reach-router";

interface NavLinkProps {
    link: string;
    text: string;
    partiallyActive?: boolean;
}

const Nav: React.FC = () => {
    const location = useLocation().pathname;

    return (
        <nav className="my-6 space-x-12 min-w-max text-center">
            <NavLink
                link="/"
                text="POSTS"
                partiallyActive={location.startsWith("/posts")}
            />
            <NavLink link="/about/" text="ABOUT" />
        </nav>
    );
};

const NavLink: React.FC<NavLinkProps> = ({
    link,
    text,
    partiallyActive,
}: NavLinkProps) => (
    <Link
        partiallyActive={partiallyActive}
        to={link}
        className="pb-2 text-black border-b hover:border-black"
        activeClassName="border-black"
    >
        {text}
    </Link>
);

export { Nav };
