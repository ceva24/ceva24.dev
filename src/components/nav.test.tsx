import React from "react";
import { render } from "@testing-library/react";
import { Nav } from "./nav";

jest.mock("@gatsbyjs/reach-router", (): unknown => ({
    ...jest.requireActual("@gatsbyjs/reach-router"),
    useLocation: jest.fn().mockImplementation(() => {
        return { pathname: "/" };
    }),
}));

describe("nav", () => {
    it("renders", () => {
        const { asFragment } = render(<Nav />);

        expect(asFragment()).toMatchSnapshot();
    });
});
