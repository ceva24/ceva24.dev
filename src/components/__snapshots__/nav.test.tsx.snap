// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav renders 1`] = `
<DocumentFragment>
  <nav
    class="my-6 space-x-12 min-w-max text-center"
  >
    <a
      aria-current="page"
      class="pb-2 text-black border-b border-transparent hover:border-black border-black"
      href="/"
    >
      POSTS
    </a>
    <a
      class="pb-2 text-black border-b border-transparent hover:border-black"
      href="/about/"
    >
      ABOUT
    </a>
  </nav>
</DocumentFragment>
`;
